src/mono/wasm/host/WebServerStartup.cs

```diff
@@ -168,41 +168,6 @@ static int GetNextRandomExcept(Range range, params int[] except)
             });
         });
 
-        applicationLifetime.ApplicationStarted.Register(() =>
-        {
-            TaskCompletionSource<ServerURLs> tcs = realUrlsAvailableTcs;
-            try
-            {
-                ICollection<string>? addresses = app.ServerFeatures
-                                                    .Get<IServerAddressesFeature>()
-                                                    ?.Addresses;
-
-                string? ipAddress = null;
-                string? ipAddressSecure = null;
-                if (addresses is not null)
-                {
-                    ipAddress = GetHttpServerAddress(addresses, secure: false);
-                    ipAddressSecure = GetHttpServerAddress(addresses, secure: true);
-                }
-
-                if (ipAddress == null)
-                    tcs.SetException(new InvalidOperationException("Failed to determine web server's IP address or port"));
-                else
-                    tcs.SetResult(new ServerURLs(ipAddress, ipAddressSecure));
-            }
-            catch (Exception ex)
-            {
-                _logger?.LogError($"Failed to get urls for the webserver: {ex}");
-                tcs.TrySetException(ex);
-                throw;
-            }
-
-            static string? GetHttpServerAddress(ICollection<string> addresses, bool secure)
-                => addresses?
-                        .Where(a => a.StartsWith(secure ? "https:" : "http:", StringComparison.InvariantCultureIgnoreCase))
-                        .Select(a => new Uri(a))
-                        .Select(uri => uri.ToString())
-                        .FirstOrDefault();
-        });
+        ServerURLsProvider.Hook(app, logger, applicationLifetime, realUrlsAvailableTcs);
```

Comment:
Why the move to an extension method? Is this meant to be used somewhere else too?
If it has to be moved, then the name needs to be descriptive.

Reply:
It is used from devserver as well. Method renamed